examples/step-36/step-36.cc

```diff
@@ -194,11 +197,19 @@ namespace Step36
     // The next step is to take care of the eigenspectrum. In this case, the
     // outputs are eigenvalues and eigenfunctions, so we set the size of the
     // list of eigenfunctions and eigenvalues to be as large as we asked for
-    // in the input file:
+    // in the input file. When using a PETScWrappers::MPI::Vector, the Vector
+    // is initialized using an IndexSet. IndexSet is used not only to resize the
+    // PETScWrappers::MPI::Vector but it also associates an index in the
+    // PETScWrappers::MPI::Vector with a degree of freedom (see step-40 for a
+    // a more detailed explanation). The function complete_index_set() creates
+    // an IndexSet where every valid index is part of the set. Note that this
+    // program can only be run sequentially and will throw an exception if used
+    // in parallel.
+    IndexSet eigenfunction_partitioning = complete_index_set(dof_handler.n_dofs ());
```

Comment:
Why not use ```dof_handler.n_locally_owned_dofs()```? It is also not a partitioning in my understanding (partitioning is for block systems while a single IndexSet is a locally_owned or locally_relevant set)